design/construct-tree.md

```diff
@@ -0,0 +1,343 @@
+
+The recurring theme is that intent based constructs are easy to get started with, take a lot of complexity away from the developer. The flip-side of this experience is that debugging and figuring out what the CDK did on your behalf can become challenging as applications come more complex. This problem will proliferate as more third-party constructs become available for consumption.
+
+ Exposing the construct tree will enrich the developer experience for CDK application construction by exposing the construct tree explicitly.
+
+
+Let’s take another look at an example of what a construct tree might look like:
+
+This is a Lambda function that has it’s handler code written as an asset:
+
+```typescript
+ new lambda.Function(this, 'HelloWorldHandler', {
+            runtime: lambda.Runtime.NODE_JS_8_10,
+            code: lambda.Code.directory('lambda'),
+            handler: 'hello.handler'
+ });
+```
+
+Ideally this simple Lambda code, will be represented by a construct tree that looks something like the following, with the ability to expand on constructs and navigate all the way down to their properties and values:
+
+```bash
+├── App
+│   ├── Stack 1
+│   │   ├── Lambda Function [L2 Construct](... properties ...)
+|   │   │   ├── IAM Role (policy information)
+|   │   │   ├── Asset Code (Lambda Handler, pointer to S3 location where Asset will reside)
+|   │   │   ├── Lambda Function (Function definition and properties)
+```
+
+## Requirements
+
+This list describes only the minimal set of requirements from this feature. After we release these building blocks, we will look into extending the model to meet other use cases that arise.
+
+1. **Format** - Publish underlying data as a `.json` file that contains the information required to render a construct tree view. The construct tree illustrated in the previous section is only a depiction of what a human readable(ish) view might look like
+2. **Constructs**
+    1. ***Nodes*** - CDK application constructs at the root and drill down to the constructs it contains (1+ cloud resources.) Initially, we will focus our attention on AWS resources, but the model is extensible to cloud components from any provider
+    2. **Types/Hierarchy** - Types and hierarchy information will contain the details around the level of abstraction (L1, L2, L3..), service(s), and resource(s) that they represent.
+    3. ***Metadata -*** Constructs will expose metadata which will include an array of objects associated with the construct.
```

Comment:
The 'Metadata' group sounds vague and it's not clear why these 3 fall under this but not 'Type'. 

I'd suggest dropping this grouping in this requirements section altogether. Perhaps something along the lines of - 
"The following information will be exposed for all instances of all CDK constructs, involved in the application. They are (a) Type & hierarchy - ... (b) source location - .... (c) ..."

Would that be better?